src/client-side-metrics/gcp-metrics-handler.ts

```diff
@@ -0,0 +1,313 @@
+   * (histograms and counters) and exports them to Google Cloud Monitoring
+   * through the provided `PushMetricExporter`.
+   *
+   * @param exporter - The `PushMetricExporter` instance to use for exporting
+   *   metrics to Google Cloud Monitoring. This exporter is responsible for
+   *   sending the collected metrics data to the monitoring backend. The provided exporter must be fully configured, for example the projectId must have been set.
+   */
+  constructor(exporter: PushMetricExporter) {
+    this.exporter = exporter;
+  }
+
+  /**
+   * Initializes the OpenTelemetry metrics instruments if they haven't been already.
+   * Creates and registers metric instruments (histograms and counters) for various Bigtable client metrics.
+   * Sets up a MeterProvider and configures a PeriodicExportingMetricReader for exporting metrics to Cloud Monitoring.
+   *
+   * @param {MonitoredResourceData} data The data that will be used to set up the monitored resource
+   * which will be provided to the exporter in every export call.
+   *
+   */
+  private initialize(data: MonitoredResourceData) {
+    if (!this.initialized) {
+      this.initialized = true;
+      const latencyBuckets = [
+        0, 0.01, 0.05, 0.1, 0.3, 0.6, 0.8, 1, 2, 3, 4, 5, 6, 8, 10, 13, 16, 20,
+        25, 30, 40, 50, 65, 80, 100, 130, 160, 200, 250, 300, 400, 500, 650,
+        800, 1000, 2000, 5000, 10000, 20000, 50000, 100000,
+      ];
+      const viewList = [
+        'operation_latencies',
+        'first_response_latencies',
+        'attempt_latencies',
+        'retry_count',
+        'server_latencies',
+        'connectivity_error_count',
+        'application_latencies',
+        'throttling_latencies',
+      ].map(
+        name =>
+          new View({
+            instrumentName: name,
+            name,
+            aggregation: name.endsWith('latencies')
+              ? Aggregation.Sum()
+              : new ExplicitBucketHistogramAggregation(latencyBuckets),
+          })
+      );
+      const meterProvider = new MeterProvider({
+        views: viewList,
+        resource: new Resources.Resource({
+          'service.name': 'Cloud Bigtable Table',
+          'cloud.provider': 'gcp',
+          'cloud.platform': 'gce_instance',
+          'cloud.resource_manager.project_id': data.projectId,
+          'monitored_resource.type': 'bigtable_client_raw',
+          'monitored_resource.project_id': data.projectId,
+          'monitored_resource.instance_id': data.instanceId,
+          'monitored_resource.table': data.table,
+          'monitored_resource.cluster': data.cluster,
+          'monitored_resource.zone': data.zone,
+        }).merge(new ResourceUtil.GcpDetectorSync().detect()),
+        readers: [
+          // Register the exporter
+          new PeriodicExportingMetricReader({
+            // Export metrics every 10 seconds. 5 seconds is the smallest sample period allowed by
+            // Cloud Monitoring.
+            exportIntervalMillis: 1_000,
```

Comment:
I assume this is the default (also across all languages). Just double checking : 0

Reply:
Thanks for catching this. Based on what I see it actually looks like we should export every 60 seconds. That is what the Python client does:

https://github.com/googleapis/python-bigtable/pull/924/files#diff-142d623e0503afa707461f3bd80d8808a2e10232038c24d7d69ce1df6bd4dbceR105

The comment should be corrected as well.